packages/teuchos/core/src/Teuchos_RCPDecl.hpp

```diff
@@ -546,6 +546,11 @@ class RCP {
   template<class T2>
   inline RCP(const RCP<T2>& r_ptr);
 
+  /** \brief Construct from another <tt>RCP<T2></tt> and from a raw pointer.
+   */
```

Comment:
It might be good to explain the assumptions about the lifetime of `ptr` in relation to the created `RCP` object or just mention the relationship to `std::shared_ptr`.